aptos-move/aptos-vm/src/move_vm_ext/respawned_session.rs

```diff
@@ -356,10 +356,27 @@ impl<'r> TResourceGroupView for ExecutorViewWithChangeSet<'r> {
 
     fn resource_group_size(
         &self,
-        _group_key: &Self::GroupKey,
+        group_key: &Self::GroupKey,
     ) -> PartialVMResult<ResourceGroupSize> {
-        // In respawned session, gas is irrelevant, so we return 0 (GroupSizeKind::None).
-        Ok(ResourceGroupSize::zero_concrete())
+        use AbstractResourceWriteOp::*;
+
+        if let Some(size) = self
+        .change_set
+        .resource_write_set()
+        .get(group_key)
+        .and_then(|write| match write {
+            WriteResourceGroup(group_write) => Some(Ok(group_write.maybe_group_op_size().unwrap_or(ResourceGroupSize::zero_combined()))),
```

Comment:
what should we return for resource_group_size() -> PartialVMResult<ResourceGroupSize> if it is deletion? 

returning zero_combined() seems like a correct thing?

Reply:
I mean in this case I don't think we can return anything else, but seems a bit weird to have Some(0) and None be treated the same? 

If write is a deletion, what is its size? If 0, maybe it would make sense to get rid of APIs which can have None as size?

@gelash I forgot, what was the reason for having optional sizes in group writes?

Reply:
It cannot be Some(0). We probably don't need it to be option there,  but that's outside of scope of this PR

Reply:
I think it makes sense to use 0 for deletion, I kinda remember some reason why it was option - maybe it was mirroring metadata or smt (does it not exist or is size 0?), but there shouldn't be any hard constraints and 0 is probably cleaner.